addon/models/instructor-group.js

```diff
@@ -1,60 +1,42 @@
 import Ember from 'ember';
 import DS from 'ember-data';
 
+const { attr, belongsTo, hasMany, Model } = DS;
 const { RSVP, computed } = Ember;
-const { Promise, map } = RSVP;
+const { map } = RSVP;
 
-export default DS.Model.extend({
-  title: DS.attr('string'),
-  school: DS.belongsTo('school', {async: true}),
-  learnerGroups: DS.hasMany('learner-group', {async: true}),
-  ilmSessions: DS.hasMany('ilm-session', {async: true}),
-  users: DS.hasMany('user', {async: true}),
-  offerings: DS.hasMany('offering', {async: true}),
-  coursesFromOfferings: computed('offerings.[]', function(){
-    return new Promise(resolve => {
-      this.get('offerings').then(offerings => {
-        map(offerings.toArray(), offering => {
-          return new Promise(resolve => {
-            offering.get('session').then(session => {
-              session.get('course').then(course => {
-                resolve(course);
-              });
-            });
-          });
-        }).then(courses => {
-          resolve(courses.uniq());
-        });
-      });
-    });
-  }),
-  coursesFromIlmSessions: computed('ilmSessions.[]', function(){
-    return new Promise(resolve => {
-      this.get('ilmSessions').then(ilmSessions => {
-        map(ilmSessions.toArray(), ilmSession => {
-          return new Promise(resolve => {
-            ilmSession.get('session').then(session => {
-              session.get('course').then(course => {
-                resolve(course);
-              });
-            });
-          });
-        }).then(courses => {
-          resolve(courses.uniq());
-        });
-      });
+export default Model.extend({
+  title: attr('string'),
+  school: belongsTo('school', {async: true}),
+  learnerGroups: hasMany('learner-group', {async: true}),
+  ilmSessions: hasMany('ilm-session', {async: true}),
+  users: hasMany('user', {async: true}),
+  offerings: hasMany('offering', {async: true}),
+
+  coursesFromOfferings: computed('offerings.[]', async function () {
+    const offerings = await this.get('offerings');
+    const courses = await map(offerings.toArray(), async ilmSession => {
```

Comment:
Doesn't make a difference in the results but `ilmSession` here should be `offering`.

Reply:
yep, copy/paste for the _lose_. thanks for catching that. fixed.